### ISSUE TITLE
create a Tiller / Repository query of all deployments that have updates available

### ISSUE THREAD

**Issue by jstrachan** [kubernetes/helm#1079]
it'd be awesome to be able to build something AppStore / iOS / Android like for apps installed via helm; where tiller/helm can query the repository for all the charts installed and list the charts that can be updated to newer versions; letting users select which ones they wish to upgrade.

Even cooler would be enabling auto-update on a per chart basis so that, say, tiller could periodically query for updates of 'auto-updated' apps and just update them. 

Advanced users could set the major/minor/patch release notifications/auto-updates. e.g. just auto-update patches or minor upgrades for chart `foo`

Then like #990 we could build a little `Updates` page in the kubernetes dashboard

**Comment by technosophos** [comment]
So you're thinking something like...

```console
$ helm list --upgradeable # or something...
RELEASE NAME           CURRENT         AVAILABLE
happy-owl       nginx-0.1.0        nginx-1.2.3
sad-panda        alpine-3.3        alpine-3.4
```

**Comment by jstrachan** [comment]
yeah, that'd be awesome! Then ideally making it available as a REST API so that we can have a UI on Kubernetes Dashboard to show available updates; and we could have an 'auto-updater' service in tiller to let configured services be auto-upgraded on the fly

**Comment by technosophos** [comment]
This is still on the possible feature list for the future.

**Comment by technosophos** [comment]
This might also be something that could be done more easily as a stand-alone plugin.

**Comment by bacongobbler** [comment]
This is a complete work-in-progress and a release has not come out yet, but you can try to get [`helm-whatup`](https://github.com/bacongobbler/helm-whatup) running while I'm away. This plugin checks against your locally cached repository index files to see if an update exists for your installed charts.

Usage (assuming you can get it to work in its half-baked state, damned godeps):

```
$ helm repo update
$ helm whatup
There is an update available for Workflow!
Installed version: v2.13.0
Available version: v2.14.0
```

**Comment by technosophos** [comment]
@bacongobbler that is awesome! Can you add a link to the docs/related.md as soon as you feel ready?

**Comment by bacongobbler** [comment]
absolutely. Sorry, haven't had much time to get back to that plugin, but if anyone wants to fix the dependency hell that is k8s.io/kubernetes, PRs are more than welcome!

**Comment by bacongobbler** [comment]
closing, please follow up at helm-whatup. :)

**Comment by fhemberger** [comment]
@bacongobbler Are you still working on it?

**Comment by bacongobbler** [comment]
@fhemberger sorry for the delay. Yes, I just released v0.1.0. You can install it via `helm plugin install https://github.com/bacongobbler/helm-whatup`. Thanks!

**Comment by dash042** [comment]
this should be reopened
